lgc/builder/ArithBuilder.cpp

```diff
@@ -217,27 +217,6 @@ Value *ArithBuilder::CreateQuantizeToFp16(Value *value, const Twine &instName) {
 // @param divisor : Divisor value
 // @param instName : Name to give instruction(s)
 Value *ArithBuilder::CreateSMod(Value *dividend, Value *divisor, const Twine &instName) {
-  if (divisor->getType()->getScalarType()->isIntegerTy(32) &&
-      getPipelineState()->getTargetInfo().getGpuWorkarounds().gfx10.disableI32ModToI16Mod) {
```

Comment:
@trenouf should I also remove the definition of "disableI32ModToI16Mod", and are there any subtleties I need to be aware of, like replacing it with a dummy bit?

Reply:
I think you can completely remove the workaround if backend behaves well now.